### PR TITLE
Modifying test-ib.sh so that ethernet devices are excluded. Also addi…

### PR DESCRIPTION
**Pull request by moes1**
Modifying test-ib.sh so that ethernet devices are excluded. Also adding a startup service that will check if the nvidia-uvm kernel module is loaded correctly, and load it if it is not. Finally, also adding some python3.8 dependencies that are necessary to make sure add-apt-repository works.

Edit: Adding a workaround for IPoIB naming to the 20.04 image. Also updating the dkms version for Centos-7.x at the request of @abhamidipati0614

### CODE REVIEW

**Review by jithinjosepkl — DISMISSED** [review]

File: ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh [hunk 1]
```diff
@@ -48,7 +63,6 @@ apt-get -y install numactl \
                    libnl-route-3-dev \
                    libsecret-1-0 \
 		   ansible \
-		   python3-pip \
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh:51]: @rafsalas19 , pls make sure this doesn't break Moneo.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh [hunk 2]
```diff
@@ -44,6 +44,10 @@ $COMMON_DIR/copy_test_file.sh
 # diable auto kernel updates
 $UBUNTU_COMMON_DIR/disable_auto_upgrade.sh
 
+cp $UBUNTU_COMMON_DIR/check_nvidia_uvm.sh .
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh:47]: Is this a problem also for the NC SKUs? If not, we can skip this change for LTS image.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh (hunk 1 above)
Comment (rafsalas19) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh:51]: @moes1 @jithinjosepkl the file doesn't look right.  Maybe the Merge didn't go as planned?

Comment (rafsalas19) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh:51]: I don't think we need sudo here.

Comment (moes1) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh:51]: We confirmed this doesn't break Moneo, and I cleaned up the duplicated lines.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh (hunk 2 above)
Comment (moes1) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh:47]: It appears to be a problem at least for Standard_NC64as_T4_v3, any other SKUs it would be better for me to check?

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh [hunk 3]
```diff
@@ -58,6 +58,11 @@ cp $UBUNTU_COMMON_DIR/check_nvidia_uvm.sh .
 $UBUNTU_COMMON_DIR/check_uvm_service.sh
 
 
+#rename IPoIB interfaces
+ibps="\nACTION==\"add\", SUBSYSTEM==\"net\", DRIVERS==\"?*\", ATTR{type}"
+ibps+="==\"32\", KERNEL==\"ib*\", NAME=\"ib0\""
+echo -e $ibps | sudo tee -a /etc/udev/rules.d/70-persistent-ipoib.rules
+
```
Comment (jeseszhang1010) [inline ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh:65]: Could you please describe the naming issue of IPoIB device in Ubuntu20.04, and why we need it? 
Also is it ensured that the IPoIB device can be precisely matched with the following rule:
`ACTION=="add", SUBSYSTEM=="net", DRIVERS=="?*", ATTR{type}=="32", KERNEL=="ib*", NAME="ib0"`?

I'd be better if we add these info into the commit msg.

File: ubuntu/common/check_uvm_service.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	
+   sudo mv check_nvidia_uvm.sh /usr/local/bin/check_nvidia_uvm.sh
+   
+   echo -e '[Unit]\n\nDescription=Runs /usr/local/bin/check_nvidia_uvm.sh\n\n' \
+           | sudo tee nvidia_uvm_startup.service
+   echo -e '[Service]\n\nExecStart=/usr/local/bin/check_nvidia_uvm.sh\n\n' \
+           | sudo tee -a nvidia_uvm_startup.service
+   echo -e '[Install]\n\nWantedBy=multi-user.target' \
+   	| sudo tee -a nvidia_uvm_startup.service
+
```
Comment (jeseszhang1010) [inline ubuntu/common/check_uvm_service.sh:13]: The EOF syntax is useful, and can be used here when working with multi-line text in Bash

File: ubuntu/common/check_nvidia_uvm.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        :
+fi
+
+verb=0
+while [ $# -gt 0 ] ; do
+	if [ $1 = "-v" ]; then
+		verb=1
+	fi
+	shift
+done
```
Comment (jeseszhang1010) [inline ubuntu/common/check_nvidia_uvm.sh:39]: Not sure if we really need this? Could you point me where this script is invoked with -v option?

Comment (moes1) [reply ubuntu/common/check_nvidia_uvm.sh:39]: We don't need it for now, but I just added it so we can run the script and have the error codes printed out if we need to debug something. I can get rid of it if you think it's confusing?

Comment (jeseszhang1010) [reply ubuntu/common/check_nvidia_uvm.sh:39]: I see. Since we are not actually invoking this script with -v option somewhere, I prefer we remove it.

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh (hunk 3 above)
Comment (jeseszhang1010) [reply ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh]: Thanks for your clarification, @moes1. 
Are we doing this renaming to unify the IPoIB interface name across Ubuntu 1804 and 2004, or the naming in ubuntu2004 (like ibP257s154062) breaks something?

Comment (moes1) [reply ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh]: I was doing the renaming only in ubuntu2004 just so the naming scheme was the same in both 1804 and 2004.

File: ubuntu/common/check_uvm_service.sh (hunk 4 above)
Comment (moes1) [reply ubuntu/common/check_uvm_service.sh]: Good idea! I rewrote this with the EOF syntax and it's much more readable now.

File: ubuntu/common/check_nvidia_uvm.sh (hunk 5 above)
Comment (moes1) [reply ubuntu/common/check_nvidia_uvm.sh:39]: Ok, removed.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh [hunk 6]
```diff
@@ -16,6 +16,16 @@ update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
...
+DISTPACK=/usr/lib/python3/dist-packages
+cp $DISTPACK/apt_pkg.cpython-36m-x86_64-linux-gnu.so $DISTPACK/apt_pkg.so
+apt-get install -y libcairo2-dev
+apt-get install -y python3-dev
+apt-get install -y libpython3.8-dev
+apt-get install -y libgirepository1.0-dev
+python3.8 -m pip install --ignore-installed PyGObject
+apt-get install -y software-properties-common
+
 apt-get update
```
[the remark this answers is not in the document]
Comment (moes1) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-hpc/install_utils.sh]: Good point, I might as well delete this line in this PR.

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh (hunk 2 above)
Comment (moes1) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh:47]: Is this alright @jithinjosepkl ?

Comment (jithinjosepkl) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh:47]: Can we pls check for the NC24r VM as well? (targeted SKU for LTS image).

**Review by jithinjosepkl — APPROVED** [review]

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh [hunk 7]
```diff
@@ -44,6 +44,7 @@ $COMMON_DIR/copy_test_file.sh
 # diable auto kernel updates
 $UBUNTU_COMMON_DIR/disable_auto_upgrade.sh
 
+
```
Comment (jeseszhang1010) [inline ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh:47]: NIT: unnecessary change.

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh [hunk 8]
```diff
@@ -54,6 +54,8 @@ $UBUNTU_COMMON_DIR/install_monitoring_tools.sh
 # diable auto kernel updates
 $UBUNTU_COMMON_DIR/disable_auto_upgrade.sh
 
+
+
 # clear history
```
Comment (jeseszhang1010) [inline ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh:59]: same comment

File: ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh (hunk 7 above)
Comment (moes1) [reply ubuntu/ubuntu-18.x/ubuntu-18.04-LTS-hpc/install.sh]: removed extra space.

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh (hunk 8 above)
Comment (moes1) [reply ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh]: removed extra spaces.

**Comment by yakovdyadkin** [comment]
> azure.azhpc-image failed https://dev.azure.com/hpc-platform-team/hpc-image-val/_build/results?buildId=3660&view=logs&j=55edb413-3e42-51ec-aec8-1bf7ffe492a6&t=6368d010-b137-5c72-4e17-cf37b77ef6d5
> 
> Seems not related to this change though. Please double check. I can approve once it's confirmed.

Just checked, failure was not related to changes. You can approve.

**Review by jeseszhang1010 — APPROVED** [review]

**Comment by jeseszhang1010** [comment]
> > azure.azhpc-image failed https://dev.azure.com/hpc-platform-team/hpc-image-val/_build/results?buildId=3660&view=logs&j=55edb413-3e42-51ec-aec8-1bf7ffe492a6&t=6368d010-b137-5c72-4e17-cf37b77ef6d5
> > Seems not related to this change though. Please double check. I can approve once it's confirmed.
> 
> Just checked, failure was not related to changes. You can approve.

Thanks for checking @yakovdyadkin. Can we re-trigger the pipeline again and let it pass? 
It's better to merge this PR after it's passed.
